python/paddle/fluid/executor.py

```diff
@@ -391,6 +393,14 @@ def _as_lodtensor(data, place):
                 ndarray to LoDTensor. Please convert data to LoDTensor \
                 directly before feeding the data.\
                 ")
+
+    #NOTE(zhiqiu): convert python builtin ,like float and int, to numpy array
+    if not isinstance(data, np.ndarray):
+        if np.isscalar(data) and dtype:
```

Comment:
dtype must be provided when `data` is a scalar?

Reply:
Yes, otherwise we can't do proper type conversion.